src/docfx/restore/RestoreFile.cs

```diff
@@ -44,15 +44,12 @@ public static async Task Restore(string url, Config config)
 
             using (InterProcessMutex.Create(filePath))
             {
-                if (!File.Exists(filePath))
-                {
-                    PathUtility.CreateDirectoryFromFilePath(filePath);
-                    File.Move(tempFile, filePath);
-                }
-                else
-                {
-                    File.Delete(tempFile);
-                }
+                PathUtility.CreateDirectoryFromFilePath(filePath);
+
+                if (File.Exists(filePath))
```

Comment:
line 49 - 52 can become a single atomic operation using `File.Replace`, it internally forwards the call to [ReplaceFileW](https://docs.microsoft.com/en-us/windows/win32/api/winbase/nf-winbase-replacefilew)
```suggestion
                File.Replace(tempFile, filePath, null);
```

Reply:
File.Replace will throw errors when dest file doesn't exists